File: scripts-investigacion/test_login.py
import requests
import json

session = requests.Session()

# Step 1: Get CSRF token
response = session.get('https://usuarios.lcnidiomas.edu.co/')
print("Cookies after GET /:", session.cookies.get_dict())

# Let's try to post to /login
login_url = 'https://usuarios.lcnidiomas.edu.co/login'

# We might need the X-XSRF-TOKEN header. It is the unquoted XSRF-TOKEN cookie.
xsrf_token = session.cookies.get('XSRF-TOKEN')
if xsrf_token:
    xsrf_token = requests.utils.unquote(xsrf_token)

headers = {
    'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36',
    'Accept': 'application/json, text/plain, */*',
    'Content-Type': 'application/json',
    'X-Requested-With': 'XMLHttpRequest'
}

if xsrf_token:
    headers['X-XSRF-TOKEN'] = xsrf_token

payload = {
    "email": "YOUR_EMAIL@example.com",
    "password": "YOUR_PASSWORD"
}

response = session.post(login_url, json=payload, headers=headers)
print("Login Status Code:", response.status_code)
try:
    print("Login Response JSON:", response.json())
except:
    print("Login Response Text (Snippet):", response.text[:500])

print("Cookies after login:", session.cookies.get_dict())

File: research/test_login.js
const fs = require('fs');

async function testLogin() {
  try {
    // Maybe we just POST to /api/login?
    const loginRes = await fetch('https://usuarios.lcnidiomas.edu.co/api/login', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json'
      },
      body: JSON.stringify({
        email: 'YOUR_EMAIL@example.com',
        password: 'YOUR_PASSWORD'
      })
    });

    console.log('Login Status:', loginRes.status);
    const loginText = await loginRes.text();
    console.log('Login Response:', loginText);
    
    // Some headers might have multiple set-cookie values. 
    // fetch API sometimes combines them, or we can iterate through them
    for (const [key, value] of loginRes.headers.entries()) {
      if (key.toLowerCase() === 'set-cookie') {
        console.log('Set-Cookie:', value);
      }
    }

  } catch (error) {
    console.error('Error:', error);
  }
}

testLogin();
